pandas/core/series.py

```diff
@@ -659,7 +674,13 @@ def _values(self):
     @Appender(base.IndexOpsMixin.array.__doc__)  # type: ignore[misc]
     @property
     def array(self) -> ExtensionArray:
-        return self._mgr._block.array_values()
+        if isinstance(self._mgr, SingleBlockManager):
+            return self._mgr._block.array_values()
+        else:
+            arr = self._mgr.array
+            if isinstance(arr, np.ndarray):
+                arr = PandasArray(arr)
+            return arr
```

Comment:
Could also move this to SingleArrayManager and add a `array_values()` method there (for SingleBlockManager this lives on the Block, not the Manager)